File: frontend/src/app/models/dailyme-alert.ts
export interface DailyMeAlert {
  id: number;
  patientId: string;
  riskLevel: 'HIGH' | 'MEDIUM' | 'LOW';
  status: 'NEW' | 'SEEN' | 'RESOLVED';
  reason?: string | null;
  source?: string | null;
  createdAt?: string | null;
  resolvedAt?: string | null;
}

File: backend/Dailyme-service/src/main/java/tn/esprit/dailymeservice/Model/DailyMeAlert.java
package tn.esprit.dailymeservice.Model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "dailyme_alerts")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class DailyMeAlert {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String patientId;

    @Column(nullable = false)
    private String riskLevel; // HIGH / MEDIUM / LOW

    @Column(nullable = false)
    private String status; // NEW / SEEN / RESOLVED

    @Column(length = 600)
    private String reason; // short message

    @Column(nullable = false)
    private String source; // e.g. "DAILYME_INSIGHTS"

    @Column(nullable = false)
    private LocalDateTime createdAt;

    private LocalDateTime resolvedAt;
}